opm/simulators/flow/NonlinearSolver.hpp

```diff
@@ -122,14 +124,15 @@ struct NonlinearSolverParameters
             , nonlinearIterationsLast_(0)
             , linearIterationsLast_(0)
             , wellIterationsLast_(0)
+            , enableAdaptiveTimeStepping_(true)
```

Comment:
this feels like a layer violation. it would, imo, be better to use the old approach, but pass the TimeStepControlInterface by ptr so you can pass nullptr from the non-adaptive case (and check for null on the inside).

Reply:
I agree.